types/lib/model.d.ts

```diff
@@ -1909,9 +1909,15 @@ export abstract class Model<T = any, T2 = any> extends Hooks {
   public static upsert<M extends Model>(
     this: { new (): M } & typeof Model,
     values: object,
-    options?: UpsertOptions
+    options?: UpsertOptions & { returning?: false | undefined }
```

Comment:
Just a thought:
what might also work: 

```ts
public static upsert<M extends Model, T extends boolean>(this: { new (): M } & typeof Model, values: object, options?: UpsertOptions): Promise<T extends true ? [M, boolean] : boolean>
```

would reduce duplication..

Reply:
@eseliger That'd be preferable indeed, most of these typings were written when this functionality wasn't around :)